### DISCUSSION TITLE
Руководство ПЛИ, тело документа, п.1.4 Секция "Витальные параметры" VITALPARAM

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#283] [Лабораторная РГ]
Секция предназначена для указания информации о витальных параметрах пациента, влияющих на интерпретацию результатов лабораторного исследования.
В данную секцию включается следующая информация:
• Дата измерения;
• Комментарии врача к собранным данным (опционально);
• Код витального параметра;
• Значение параметра (при наличии информации);
• Латеральность (опционально);
• Порядковый номер (опционально);
• Референтный интервал (опционально).

**Comment by eugenekogan-spb** [comment]
<img width="763" alt="image" src="https://github.com/user-attachments/assets/7147fb0e-9aec-4dc1-b4f8-3e2f737fd45c">
<img width="766" alt="image" src="https://github.com/user-attachments/assets/858e0743-525c-40e9-988a-48b1f5192dd4">
<img width="764" alt="image" src="https://github.com/user-attachments/assets/d4d28ade-2bc8-4eef-8e6b-2cc4404a5884">

**Comment by eugenekogan-spb** [comment]
У3-3: Секция "Витальные параметры" ОБЯЗАНА иметь хотя бы один [1..*] элемент
entry, содержащий сведения о витальном параметре, который:
1. ОБЯЗАН иметь один [1..1] элемент organizer (атрибут @classCode должен иметь значение "CLUSTER", атрибут @moodCode должен иметь значение "EVN"), который:
1.1. ОБЯЗАН иметь один [1..1] элемент statusCode (атрибут @code должен иметь значение "completed"), содержащий статус измерения;
1.2. ОБЯЗАН иметь один [1..1] элемент effectiveTime (атрибут @value должен быть представлен с не пустым значением), содержащий дату измерения;
1.3. МОЖЕТ иметь один [0..1] элемент precondition (атрибут @typeCode должен иметь значение "PRCN"), содержащий комментарии врача к собранным данным, который:
1.3.1. ОБЯЗАН иметь один [1..1] элемент criterion, который:
1.3.1.1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @code должен иметь значение "ASSERTION", атрибут @codeSystem должен иметь значение "2.16.840.1.113883.5.4");
1.3.1.2. ОБЯЗАН иметь один [1..1] элемент value (атрибут @xsi:type должен иметь значение "ST"), содержащий комментарии врача к собранным данным, который должен иметь не пустое наполнение;

**Reply by eugenekogan-spb** [reply]
Перечисленные здесь атрибуты могут быть представлены ресурсом Observation c атрибутами:
```
* status = #final

* effectiveDateTime = "2022-11-15T08:30:00+03:00"

* note 
  * text = "данные записаны со слов медсестры"
```

**Reply by eugenekogan-spb** [reply]
Вопрос: как показать, какое отношение имеет наш витальный параметр к результату исследования?
Возможно, следует включить его в состав DiagnosticReport:

```
instanceOf: DiagnosticReport
* supportingInfo
  * type
    * text = "supporting clinical information"
    * coding
      * system = "http://terminology.hl7.org/CodeSystem/v2-0936"
      * code = #SCI
  * reference = Reference (observation-vital-sign-diuresis)
```

**Reply by maximalexeev** [reply]
Возможно, стоит вводить какой-то служебный раздел с направлением, под крышей ServiceRequest, пока нам СДА направления не завезли. И вот в этот SR складывать обсервы с витальными параметрами, ибо это все же данные направлкения, а не результата. При выполнении ЛИ вес и рост пациента не измеряют же

**Reply by eugenekogan-spb** [reply]
Решили: ServiceRequest.supportingInfo, потому что авторство этой информации - в заказывающей организации.

**Reply by eugenekogan-spb** [reply]
//для R4 - reference(Any)

```
instanceOf: ServiceRequest
* supportingInfo = Reference (observation-vital-sign-diuresis)
```

//для R5 - codeableReference
```
instanceOf: ServiceRequest
* supportingInfo[+]
  * concept
    * text = "supporting clinical information"
    * coding
      * system = "http://terminology.hl7.org/CodeSystem/v2-0936"
      * code = #SCI
  * reference = Reference (observation-vital-sign-diuresis)
```

**Reply by eugenekogan-spb** [reply]
Надо будет учесть при описании ServiceRequest @OlegPenzin

**Comment by eugenekogan-spb** [comment]
2. Компоненты

**компоненты, ST**

1.4. ОБЯЗАН иметь от одного до двух [1..2] элементов component (атрибут @typeCode должен иметь значение "COMP"), содержащих витальный параметр, который:
1.4.1.ОБЯЗАН иметь один [1..1] элемент observation (атрибут @classCode должен иметь значение "OBS", атрибут @moodCode должен иметь значение "EVN"), который:
1.4.1.1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.262", атрибут @codeSystemName должен иметь значение "Витальные параметры", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий код витального параметра;
1.4.1.2. ДОЛЖЕН иметь один [1..1] элемент value (атрибут @value/@xsi:type должен иметь значение по справочнику "Витальные параметры" (OID:1.2.643.5.1.13.13.99.2.262), поле "Код типа данных"), содержащий значение параметра.
ЕСЛИ атрибут value/@xsi:type имеет значение "ST", ТО элемент должен иметь не пустое наполнение;

**Reply by eugenekogan-spb** [reply]
Комментарий: в справочнике МЗ РФ [код типа данных ST отсутствует](https://fnsi2.test.egisz.rosminzdrav.ru/dictionaries/1.2.643.5.1.13.13.99.2.262/passport/8.5).

**Reply by eugenekogan-spb** [reply]
Здесь, видимо, нотации различаются. 

1) СЭМД обязывает создать двухуровневую структуру, в которой код и значение измеренного параметра находятся на втором уровне. FHIR позволяет простые наблюдения отражать одним уровнем, а имеющие сложную структуру - двумя. Тем не менее, нотации обеспечивают взаимно-однозначное соответствие.

В случае простого наблюдения, как в примере, приведенном в СЭМД (объем диуреза) ограничиваемся одним уровнем.
```
* code
  * text = "Суммарный диурез"
  * coding
    * system = "1.2.643.5.1.13.13.99.2.262"
    * code = #8
```
2) Код типа данных, требуемый к передаче в СЭМД, видимо, определяется типом данных передаваемого значения (это догадка, в руководстве явно не указано).
ST - string
PQ - quantity
BL - boolean
CD - code

В Observation для различных типов данных имеются различные value [x]. Предлагается дополнительно тип данных в FHIR не указывать, и в случае преобразования в CDA брать указанное значение из справочника МЗ РФ.

**Comment by eugenekogan-spb** [comment]
**BL**

ЕСЛИ атрибут value/@xsi:type имеет значение "BL", ТО атрибут value/@value должен иметь значение "true" или "false";

**Comment by eugenekogan-spb** [comment]
**PQ**

ЕСЛИ атрибут value/@xsi:type имеет значение "PQ", ТО атрибут value/@value должен быть представлен со значением измеренного параметра, атрибут value/@unit должен быть представлен со значением из справочника "Единицы измерения" (OID: 1.2.643.5.1.13.13.11.1358), поле "Код UCUM";

**Reply by eugenekogan-spb** [reply]
В этом случае значение передается в атрибуте valueQuantity
```
* valueQuantity
  * value = 1500
  * unit = "мл"
  * system = "1.2.643.5.1.13.13.11.1358"
  * code = #16
```

Вопрос: как передать версию справочника единиц измерения?

**Comment by eugenekogan-spb** [comment]
**CD**

ЕСЛИ атрибут value/@xsi:type имеет значение "CD", ТО атрибут value/@codeSystem должен быть представлен со значением из справочника "Витальные параметры" (OID:1.2.643.5.1.13.13.99.2.262), поле "Справочник", атрибуты value/@code, value/@displayName, value/@codeSystemName, value/@codeSystemVersion должны быть представлены с не пустыми значениями.
Данный элемент:
1. МОЖЕТ иметь один [0..1] элемент translation (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1358", атрибут @codeSystemName должен иметь значение "Единицы измерения", атрибуты @value, @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями; атрибут @value должен иметь значение измеряемого параметра), содержащий единицы измерения параметра. Возможны следующие альтернативы:
- ЕСЛИ атрибут value/@xsi:type имеет значение "PQ", ТО элемент обязан быть представлен;
- ЕСЛИ иначе, ТО элемент должен отсутствовать;

**Comment by eugenekogan-spb** [comment]
**латеральность**

3.МОЖЕТ иметь один [0..1] элемент entryRelationship (атрибут @typeCode должен иметь значение "COMP"), содержащий латеральность, который:
3.1. ОБЯЗАН иметь один [1..1] элемент observation (атрибут @classCode должен иметь значение "OBS", атрибут @moodCode должен иметь значение "EVN"), который:
3.1.1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @code должен иметь значение "6034", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.166", атрибут @codeSystemName должен иметь значение "Кодируемые поля CDA документов", атрибут @displayName должен иметь значение "Латеральность", атрибут @codeSystemVersion должен быть представлен с не пустым значением), содержащий код поля;
3.1.2. ОБЯЗАН иметь один [1..1] элемент value (атрибут @xsi:type должен иметь значение "CD", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.778", атрибут @codeSystemName должен иметь значение "Латеральность", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий латеральность;

**Reply by eugenekogan-spb** [reply]
Документация FHIR [требует не использовать Observation для передачи информации, иной чем результаты наблюдений](http://hl7.org/fhir/observation.html#bnr). Следовательно, латеральность необходимо передать в специфическом атрибуте, например, Observation.bodyStructure.includedStructure.laterality

**Comment by eugenekogan-spb** [comment]
**порядковый номер**

4.МОЖЕТ иметь один [0..1] элемент entryRelationship (атрибут @typeCode должен иметь значение "COMP"), содержащий порядковый номер, который:
4.1. ОБЯЗАН иметь один [1..1] элемент observation (атрибут @classCode должен иметь значение "OBS", атрибут @moodCode должен иметь значение "EVN"), который:
4.1.1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @code должен иметь значение "8034", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.166", атрибут @codeSystemName должен иметь значение "Кодируемые поля CDA документов", атрибут @displayName должен иметь значение "Порядковый номер", атрибут @codeSystemVersion должен быть представлен с не пустым значением), содержащий код поля;
4.1.2. ОБЯЗАН иметь один [1..1] элемент value (атрибут @xsi:type должен иметь значение "REAL", атрибут @value должен быть представлен с не пустым значением), содержащий порядковый номер;

**Reply by eugenekogan-spb** [reply]
Вопрос: можно ли взять порядковый номер из массива Observation, или надо передавать его специфическим полем?

**Reply by eugenekogan-spb** [reply]
Согласно обсуждению на встрече 05.11.24: неочевидна цель упорядочивая списка витальных параметров. Но поскольку требование есть, и с технической точки зрения массив объектов не упорядочен, необходим специальное указание порядкового номера. Предлагается сделать это с помощью идентификатора. 

```
* identifier[+]
  * type
    * text = "rank"
  * system = "numbers"
  * value = "1"
```

**Reply by eugenekogan-spb** [reply]
С другой стороны, при использовании идентификатора получаются строковые значения. Возможно, более корректно использовать специальное расширение с числовым значением.

```
* extension[+]
  * url = "rank"
  * valuePositiveInt = 1
```

**Comment by eugenekogan-spb** [comment]
**референтный интервал**

5.МОЖЕТ иметь один [0..1] элемент referenceRange, содержащий референтный интервал, который:
5.1. ОБЯЗАН иметь один [1..1] элемент observationRange, который:
5.1.1. ОБЯЗАН иметь один [1..1] элемент text, содержащий описание значения референтного интервала, который должен иметь не пустое наполнение;
5.1.2.МОЖЕТ иметь один [0..1] элемент value (атрибут @xsi:type должен иметь значение "IVL_PQ"), содержащий границы референтного интервала, который:
5.1.2.1. ДОЛЖЕН иметь один [1..1] элемент low (атрибуты @value, @unit должны быть представлены с не пустыми значениями; атрибут @value должен иметь значение параметра, атрибут @unit должен иметь значение по справочнику "Единицы измерения" (OID: 1.2.643.5.1.13.13.11.1358), поле "Код UCUM"), содержащий нижняя граница референтного интервала, который:
5.1.2.1.1.ОБЯЗАН иметь один [1..1] элемент translation (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1358", атрибут @codeSystemName должен иметь значение "Единицы измерения", атрибуты @value, @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями; атрибут @value должен иметь значение параметра), содержащий единицы измерения;
5.1.2.2. ДОЛЖЕН иметь один [1..1] элемент high (атрибуты @value, @unit должны быть представлены с не пустыми значениями; атрибут @value должен иметь значение параметра, атрибут @unit должен иметь значение по справочнику "Единицы измерения" (OID: 1.2.643.5.1.13.13.11.1358), поле "Код UCUM"), содержащий верхняя граница референтного интервала, который:
5.1.2.2.1.ОБЯЗАН иметь один [1..1] элемент translation (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1358", атрибут @codeSystemName должен иметь значение "Единицы измерения", атрибуты @value, @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями; атрибут @value должен иметь значение параметра), содержащий единицы измерения.

**Reply by eugenekogan-spb** [reply]
Референтный интервал может быть передан стандартными атрибутами ресурса Observation
```
* referenceRange
  * low
    * value = 1000
    * unit = "мл"
    * system = "1.2.643.5.1.13.13.11.1358"
    * code = #16
  * high
    * value = 2000
    * unit = "мл"
    * system = "1.2.643.5.1.13.13.11.1358"
    * code = #16
```
